### ISSUE TITLE
[Bug]: [benchmark][cluster] queryNode panic `what():  Error:RemoveDir:Directory not empty` in the scenario of serial load-release collection which replica=2

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#33591]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.4-20240603-d6a99403-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0rc66
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo task: multi-vector-corn-1-1717423200
test case name: test_hybrid_search_locust_load_release_replica_cluster

server:
```
[2024-06-04 04:24:55,329 -  INFO - fouram]: [Base] Deploy initial state: 
I0603 14:10:27.771329     402 request.go:665] Waited for 1.180191622s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.local/apis/milvus.io/v1beta1?timeout=32s
I0603 14:10:37.970694     402 request.go:665] Waited for 6.994142787s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.local/apis/apps/v1?timeout=32s
NAME                                                              READY   STATUS             RESTARTS           AGE     IP              NODE         NOMINATED NODE   READINESS GATES
multi-vector-corn-1-1717423200-2-etcd-0                           1/1     Running            0                  8m38s   10.104.20.103   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-etcd-1                           1/1     Running            0                  8m37s   10.104.15.142   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-etcd-2                           1/1     Running            0                  8m37s   10.104.25.144   4am-node30   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-datacoord-c5577dc46lj5z   1/1     Running            3 (7m27s ago)      8m39s   10.104.6.247    4am-node13   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-datanode-5c44b846c5svws   1/1     Running            4 (2m38s ago)      8m38s   10.104.30.155   4am-node38   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-indexcoord-544c6f87qxtv   1/1     Running            0                  8m38s   10.104.14.59    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-indexnode-798dc9ccswn5v   1/1     Running            3 (7m27s ago)      8m39s   10.104.6.246    4am-node13   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-proxy-84cdc6c8f-cfxwp     1/1     Running            5 (2m8s ago)       8m39s   10.104.14.56    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querycoord-64bd8658w4bb   1/1     Running            5 (2m8s ago)       8m39s   10.104.14.54    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querynode-d65545d5lg5xr   1/1     Running            4 (6m56s ago)      8m38s   10.104.14.61    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querynode-d65545d5zxrls   1/1     Running            4 (6m57s ago)      8m38s   10.104.13.254   4am-node16   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-rootcoord-6444b8ffz95js   1/1     Running            5 (2m8s ago)       8m39s   10.104.14.55    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-0                          1/1     Running            0                  8m37s   10.104.17.49    4am-node23   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-1                          1/1     Running            0                  8m36s   10.104.16.176   4am-node21   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-2                          1/1     Running            0                  8m36s   10.104.15.151   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-3                          1/1     Running            0                  8m35s   10.104.26.34    4am-node32   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-0                  1/1     Running            0                  8m37s   10.104.20.105   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-1                  1/1     Running            0                  8m37s   10.104.17.50    4am-node23   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-2                  1/1     Running            0                  8m36s   10.104.15.152   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-init-b2862         0/1     Completed          0                  8m39s   10.104.14.57    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-broker-0                  1/1     Running            0                  8m38s   10.104.14.60    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-proxy-0                   1/1     Running            0                  8m37s   10.104.20.75    4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-pulsar-init-mxfm7         0/1     Completed          0                  8m39s   10.104.14.58    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-recovery-0                1/1     Running            0                  8m38s   10.104.20.71    4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-0               1/1     Running            0                  8m38s   10.104.20.104   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-1               1/1     Running            0                  7m11s   10.104.26.62    4am-node32   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-2               1/1     Running            0                  4m31s   10.104.15.188   4am-node20   <none>           <none> (base.py:258)
[2024-06-04 04:24:55,329 -  INFO - fouram]: [Cmd Exe]  kubectl get pods  -n qa-milvus  -o wide | grep -E 'NAME|multi-vector-corn-1-1717423200-2-milvus|multi-vector-corn-1-1717423200-2-minio|multi-vector-corn-1-1717423200-2-etcd|multi-vector-corn-1-1717423200-2-pulsar|multi-vector-corn-1-1717423200-2-zookeeper|multi-vector-corn-1-1717423200-2-kafka|multi-vector-corn-1-1717423200-2-log|multi-vector-corn-1-1717423200-2-tikv'  (util_cmd.py:14)
[2024-06-04 04:25:11,565 -  INFO - fouram]: [CliClient] pod details of release(multi-vector-corn-1-1717423200-2): 
 I0604 04:24:56.584241     501 request.go:665] Waited for 1.165364896s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.local/apis/cert-manager.io/v1?timeout=32s
I0604 04:25:06.783651     501 request.go:665] Waited for 6.995815868s due to client-side throttling, not priority and fairness, request: GET:https://kubernetes.default.svc.cluster.local/apis/locust.io/v1?timeout=32s
NAME                                                              READY   STATUS             RESTARTS           AGE     IP              NODE         NOMINATED NODE   READINESS GATES
multi-vector-corn-1-1717423200-2-etcd-0                           1/1     Running            0                  14h     10.104.20.103   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-etcd-1                           1/1     Running            0                  14h     10.104.15.142   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-etcd-2                           1/1     Running            0                  14h     10.104.25.144   4am-node30   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-datacoord-c5577dc46lj5z   1/1     Running            3 (14h ago)        14h     10.104.6.247    4am-node13   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-datanode-5c44b846c5svws   1/1     Running            4 (14h ago)        14h     10.104.30.155   4am-node38   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-indexcoord-544c6f87qxtv   1/1     Running            0                  14h     10.104.14.59    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-indexnode-798dc9ccswn5v   1/1     Running            3 (14h ago)        14h     10.104.6.246    4am-node13   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-proxy-84cdc6c8f-cfxwp     1/1     Running            5 (14h ago)        14h     10.104.14.56    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querycoord-64bd8658w4bb   1/1     Running            5 (14h ago)        14h     10.104.14.54    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querynode-d65545d5lg5xr   0/1     Running            23 (33s ago)       14h     10.104.14.61    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-querynode-d65545d5zxrls   1/1     Running            50 (44m ago)       14h     10.104.13.254   4am-node16   <none>           <none>
multi-vector-corn-1-1717423200-2-milvus-rootcoord-6444b8ffz95js   1/1     Running            5 (14h ago)        14h     10.104.14.55    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-0                          1/1     Running            0                  14h     10.104.17.49    4am-node23   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-1                          1/1     Running            0                  14h     10.104.16.176   4am-node21   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-2                          1/1     Running            0                  14h     10.104.15.151   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-minio-3                          1/1     Running            0                  14h     10.104.26.34    4am-node32   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-0                  1/1     Running            0                  14h     10.104.20.105   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-1                  1/1     Running            0                  14h     10.104.17.50    4am-node23   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-2                  1/1     Running            0                  14h     10.104.15.152   4am-node20   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-bookie-init-b2862         0/1     Completed          0                  14h     10.104.14.57    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-broker-0                  1/1     Running            0                  14h     10.104.14.60    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-proxy-0                   1/1     Running            0                  14h     10.104.20.75    4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-pulsar-init-mxfm7         0/1     Completed          0                  14h     10.104.14.58    4am-node18   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-recovery-0                1/1     Running            0                  14h     10.104.20.71    4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-0               1/1     Running            0                  14h     10.104.20.104   4am-node22   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-1               1/1     Running            0                  14h     10.104.26.62    4am-node32   <none>           <none>
multi-vector-corn-1-1717423200-2-pulsar-zookeeper-2               1/1     Running            0                  14h     10.104.15.188   4am-node20   <none>           <none>
```

<img width="1161" alt="截屏2024-06-04 14 32 46" src="https://github.com/milvus-io/milvus/assets/26307815/c849d832-0360-4a2e-9cb3-334989350132">
<img width="1286" alt="截屏2024-06-04 14 33 07" src="https://github.com/milvus-io/milvus/assets/26307815/9d6d98a3-1247-418c-b896-3510f58da609">

client pod name: multi-vector-corn-1-1717423200-3201708958
test result:
```
[2024-06-04 04:24:33,218 -  INFO - fouram]: Print locust final stats. (locust_runner.py:56)
[2024-06-04 04:24:33,218 -  INFO - fouram]: Type     Name                                                                          # reqs      # fails |    Avg     Min     Max    Med |   req/s  failures/s (stats.py:789)
[2024-06-04 04:24:33,218 -  INFO - fouram]: --------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|----------- (stats.py:789)
[2024-06-04 04:24:33,218 -  INFO - fouram]: grpc     load_release                                                                     512     0(0.00%) |  84367    1629  444984  80000 |    0.01        0.00 (stats.py:789)
[2024-06-04 04:24:33,219 -  INFO - fouram]: --------|----------------------------------------------------------------------------|-------|-------------|-------|-------|-------|-------|--------|----------- (stats.py:789)
[2024-06-04 04:24:33,219 -  INFO - fouram]:          Aggregated                                                                       512     0(0.00%) |  84367    1629  444984  80000 |    0.01        0.00 (stats.py:789)
[2024-06-04 04:24:33,219 -  INFO - fouram]:  (stats.py:790)
[2024-06-04 04:24:33,223 -  INFO - fouram]: [PerfTemplate] Report data: 
{'server': {'deploy_tool': 'helm',
            'deploy_mode': 'cluster',
            'config_name': 'cluster_2c8m',
            'config': {'queryNode': {'resources': {'limits': {'cpu': '8.0',
                                                              'memory': '32Gi'},
                                                   'requests': {'cpu': '5.0',
                                                                'memory': '17Gi'}},
                                     'replicas': 2},
                       'indexNode': {'resources': {'limits': {'cpu': '8.0',
                                                              'memory': '8Gi'},
                                                   'requests': {'cpu': '5.0',
                                                                'memory': '5Gi'}},
                                     'replicas': 1},
                       'dataNode': {'resources': {'limits': {'cpu': '2.0',
                                                             'memory': '8Gi'},
                                                  'requests': {'cpu': '2.0',
                                                               'memory': '5Gi'}}},
                       'cluster': {'enabled': True},
                       'pulsar': {},
                       'kafka': {},
                       'minio': {'metrics': {'podMonitor': {'enabled': True}}},
                       'etcd': {'metrics': {'enabled': True,
                                            'podMonitor': {'enabled': True}}},
                       'metrics': {'serviceMonitor': {'enabled': True}},
                       'log': {'level': 'debug'},
                       'image': {'all': {'repository': 'harbor.milvus.io/milvus/milvus',
                                         'tag': '2.4-20240603-d6a99403-amd64'}}},
            'host': 'multi-vector-corn-1-1717423200-2-milvus.qa-milvus.svc.cluster.local',
            'port': '19530',
            'uri': ''},
 'client': {'test_case_type': 'ConcurrentClientBase',
            'test_case_name': 'test_hybrid_search_locust_load_release_replica_cluster',
            'test_case_params': {'dataset_params': {'metric_type': 'L2',
                                                    'dim': 128,
                                                    'scalars_index': {'id': {},
                                                                      'int64_1': {'index_type': 'INVERTED'},
                                                                      'varchar_1': {'index_type': 'INVERTED'}},
                                                    'vectors_index': {'float_vector_1': {'index_type': 'HNSW',
                                                                                         'index_param': {'M': 8,
                                                                                                         'efConstruction': 200},
                                                                                         'metric_type': 'L2'},
                                                                      'float_vector_2': {'index_type': 'DISKANN',
                                                                                         'index_param': {},
                                                                                         'metric_type': 'IP'},
                                                                      'float_vector_3': {'index_type': 'IVF_SQ8',
                                                                                         'index_param': {'nlist': 2048},
                                                                                         'metric_type': 'L2'}},
                                                    'scalars_params': {'float_vector_1': {'params': {'dim': 128},
                                                                                          'other_params': {'dataset': 'sift',
                                                                                                           'dim': 128}},
                                                                       'float_vector_2': {'params': {'dim': 128},
                                                                                          'other_params': {'dataset': 'sift',
                                                                                                           'dim': 128}},
                                                                       'float_vector_3': {'params': {'dim': 128},
                                                                                          'other_params': {'dataset': 'sift',
                                                                                                           'dim': 128}}},
                                                    'dataset_name': 'sift',
                                                    'dataset_size': 5000000,
                                                    'ni_per': 10000},
                                 'collection_params': {'other_fields': ['float_vector_1',
                                                                        'float_vector_2',
                                                                        'float_vector_3',
                                                                        'int64_1',
                                                                        'varchar_1'],
                                                       'shards_num': 2},
                                 'load_params': {'replica_number': 2},
                                 'resource_groups_params': {'reset': False},
                                 'database_user_params': {'reset_rbac': False,
                                                          'reset_db': False},
                                 'index_params': {'index_type': 'IVF_FLAT',
                                                  'index_param': {'nlist': 1024}},
                                 'concurrent_params': {'concurrent_number': 1,
                                                       'during_time': '12h',
                                                       'interval': 20,
                                                       'spawn_rate': None},
                                 'concurrent_tasks': [{'type': 'load_release',
                                                       'weight': 1,
                                                       'params': {'replica_number': 2,
                                                                  'timeout': 600}}]},
            'run_id': 2024060333211462,
            'datetime': '2024-06-03 14:02:01.331580',
            'client_version': '2.4.0'},
 'result': {'test_result': {'index': {'RT': 2710.8078,
                                      'float_vector_1': {'RT': 2584.1561},
                                      'float_vector_2': {'RT': 1274.015},
                                      'float_vector_3': {'RT': 692.8272},
                                      'id': {'RT': 1.0185},
                                      'int64_1': {'RT': 1.0185},
                                      'varchar_1': {'RT': 2.0396}},
                            'insert': {'total_time': 665.1986,
                                       'VPS': 7516.5522,
                                       'batch_time': 1.3304,
                                       'batch': 10000},
                            'flush': {'RT': 2.5321},
                            'load': {'RT': 70.4143},
                            'Locust': {'Aggregated': {'Requests': 512,
                                                      'Fails': 0,
                                                      'RPS': 0.01,
                                                      'fail_s': 0.0,
                                                      'RT_max': 444984.06,
                                                      'RT_avg': 84367.73,
                                                      'TP50': 80000.0,
                                                      'TP99': 212000.0},
                                       'load_release': {'Requests': 512,
                                                        'Fails': 0,
                                                        'RPS': 0.01,
                                                        'fail_s': 0.0,
                                                        'RT_max': 444984.06,
                                                        'RT_avg': 84367.73,
                                                        'TP50': 80000.0,
                                                        'TP99': 212000.0}}}}}
```

### Steps To Reproduce

```markdown
concurrent test and calculation of RT and QPS

        :purpose:  `load -> release collection, replica=2`
            verify load -> release collection scenario,
            which has 4 vector fields(IVF_FLAT, HNSW, DISKANN, IVF_SQ8) and scalar fields: `int64_1`, `varchar_1`

        :test steps:
            1. create collection with fields:
                'float_vector': 128dim,
                'float_vector_1': 128dim,
                'float_vector_2': 128dim,
                'float_vector_3': 128dim,
                scalar field: int64_1, varchar_1
            2. build indexes:
                IVF_FLAT: 'float_vector'
                HNSW: 'float_vector_1',
                DISKANN: 'float_vector_2'
                IVF_SQ8: 'float_vector_3'
                INVERTED: 'int64_1', 'varchar_1'
                default scalar index: 'id'
            3. insert 5 million data
            4. flush collection
            5. build indexes again using the same params
            6. load collection
                replica: 2
            7. concurrent request: (concurrent_number=1)
                - load_release
```

**Comment by weiliu1031** [comment]
same problems with #30857

**Comment by chyezh** [comment]
should be fixed at 2.4.5, please verify it.

**Comment by wangting0128** [comment]
### verification passed

image：2.4-20240614-5fc1370f-amd64
argo task: multi-vector-corn-1-1718546400
